functional_tests/functional_test.go

```diff
@@ -611,6 +660,180 @@ var _ = ginkgo.Describe("Tests `appgw.ConfigBuilder`", func() {
 			},
 		}
 
+		ingressWithDefaultBackend := &v1beta1.Ingress{
+			Spec: v1beta1.IngressSpec{
+				Rules: []v1beta1.IngressRule{
+					{
+						Host: tests.Host,
+						IngressRuleValue: v1beta1.IngressRuleValue{
+							HTTP: &v1beta1.HTTPIngressRuleValue{
+								Paths: []v1beta1.HTTPIngressPath{
+									{
+										Path: "/A/",
+										Backend: v1beta1.IngressBackend{
+											ServiceName: serviceNameA,
+											ServicePort: intstr.IntOrString{
+												Type:   intstr.Int,
+												IntVal: 80,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				TLS: []v1beta1.IngressTLS{
+					{
+						Hosts: []string{
+							tests.Host,
+						},
+						SecretName: tests.NameOfSecret,
+					},
+				},
+				Backend: &v1beta1.IngressBackend{
+					ServiceName: serviceNameB,
+					ServicePort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 80,
+					},
+				},
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					annotations.IngressClassKey: annotations.ApplicationGatewayIngressClass,
+				},
+				Namespace: tests.Namespace,
+				Name:      tests.Name,
+			},
+		}
+
+		ingressWithDefaultHttpsBackend := &v1beta1.Ingress{
+			Spec: v1beta1.IngressSpec{
+				TLS: []v1beta1.IngressTLS{
+					{
+						Hosts: []string{
+							tests.Host,
+						},
+						SecretName: tests.NameOfSecret,
+					},
+				},
+				Backend: &v1beta1.IngressBackend{
+					ServiceName: serviceNameD,
+					ServicePort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 443,
+					},
+				},
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					annotations.IngressClassKey:             annotations.ApplicationGatewayIngressClass,
+					annotations.AppGwTrustedRootCertificate: "root-cert",
+					annotations.BackendProtocolKey:          "https",
+					annotations.FirewallPolicy:              "waf-policy",
+				},
+				Namespace: tests.Namespace,
+				Name:      tests.Name,
+			},
+		}
+
+		ingressWithDefaultBackendNoMatchedRule := &v1beta1.Ingress{
+			Spec: v1beta1.IngressSpec{
+				Rules: []v1beta1.IngressRule{
+					{
+						IngressRuleValue: v1beta1.IngressRuleValue{
+							HTTP: &v1beta1.HTTPIngressRuleValue{
+								Paths: []v1beta1.HTTPIngressPath{
+									{
+										Path: "/Z/",
+										Backend: v1beta1.IngressBackend{
+											ServiceName: serviceNameZ,
```

Comment:
Just confirming that `serviceNameZ` is deliberately missing right ?

Reply:
> Just confirming that `serviceNameZ` is deliberately missing right ?

Yes, you are right! means no match from backend